pytorch_lightning/core/step_result.py

```diff
@@ -219,7 +219,11 @@ def __set_meta(
         _internal['_reduce_on_epoch'] = max(_internal['_reduce_on_epoch'], on_epoch)
 
     def track_batch_size(self, batch):
-        batch_size = self.unpack_batch_size(batch)
+        try:
+            batch_size = self.unpack_batch_size(batch)
+        except RecursionError as re:
+            batch_size = 1
```

Comment:
@williamFalcon I'm not too sure about the correctness of setting this to 1, or of using `len(sample)` in `unpack_batch_size()` when `sample` is a string. This way, the batch size will always be 1, even we have a list of strings with `batch_size>1` coming from the dataloader. You can try to add a `batch_size` to the dataloader in the test, and `unpack_batch_size` will still indicate 1 as the batch size with the current patch.